string_manipulation.rb

```diff
@@ -6,7 +6,7 @@ def string_reverse(my_string)
   end
   i = 0
   i2 = my_string.length-1
-  my_string = my_string.split('')
+  my_string = my_string.split('') # converting into a literal array of characters to allow for modification

```

Comment:
Using split increases your space complexity to O(n). You should be able to do the indexing without using split.

Reply:
Also the swap approach below only works if the values are limited in range and will not result in an integer overflow. It's safer to swap using a temporary variable like so:
`temp = my_string[i]`
`my_string[i] = my_string[i2]`
`my_string[i2] = temp`

Reply:
Finally, none of these methods should be returning a string since the string is passed by reference and is getting modified in place.